src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -190,7 +190,8 @@ parseTypeClassDeclaration = do
 
 parseConstraint :: TokenParser Constraint
 parseConstraint = Constraint <$> parseQualified properName
-                             <*> P.many (noWildcards parseTypeAtom)
+                             <*> (P.many (noWildcards (parens parsePolyType)
```

Comment:
It was for this case:
````
-- Doesn't get parsed correctly, returns a "Unknown type Show" exception
instance dontShowMyType :: Fail (TypeConcat (TypeString a) "bla") => Show (MyType a) where
  show _ = ""

-- Gets correctly parsed, notice the double parens
instance dontShowMyType :: Fail ((TypeConcat (TypeString a) "bla")) => Show (MyType a) where
  show _ = ""
````
I mentioned it also in the first message on this PR, I'm not sure if it actually is doing anything wrong, but I agree that it does look ugly.